### ISSUE TITLE
Design the Rest API

### ISSUE THREAD

**Issue by acrisci** [dep-labs/greta#1]
The design for the Rest API should be able to support the web front end as well as be useful on its own for monitoring software like Nagios.

What endpoints should the Rest API have?

I will write a proposal for the API and post it here.

**Comment by vijayaraj210** [comment]
are you talking about Kafka API thats fronts the JMX  https://engineering.linkedin.com/apache-kafka/burrow-kafka-consumer-monitoring-reinvented

**Comment by vijayaraj210** [comment]
just saw greta, got it

**Comment by acrisci** [comment]
I haven't delved very deeply into Burrow, but I'll take a look at it. Thanks.

What I'm imaginine is a way to look at consumer performance over time through the JMX endpoint and store that information in a database. Then Greta will present it to the user in nice charts and graphs that can be updated in real time and exported into reports. Sort of like [Kabana](https://www.elastic.co/products/kibana), but for Kafka.

Maybe even the ability to automatically run performance tests and generate a report from that.

Basically a web frontend to the included Kafka scripts.
